lib/sparqlAlgebra.ts

```diff
@@ -354,10 +354,16 @@ function simplifyPath(subject: RDF.Quad_Subject, predicate: Algebra.Operation, o
 
     if (predicate.type === types.SEQ)
     {
-        let v = generateFreshVar();
-        let left = simplifyPath(subject, (<Algebra.Seq>predicate).left, v);
-        let right = simplifyPath(v, (<Algebra.Seq>predicate).right, object);
-        return left.concat(right);
+        let joiner: RDF.Quad_Subject | RDF.Variable = subject;
+        const seq: Algebra.Seq = <Algebra.Seq> predicate;
+        return Util.flatten(seq.input.map((subOp, i) => {
+            const nextJoiner = i === seq.input.length - 1 ? object : generateFreshVar();
+            const simplifiedPath = simplifyPath(joiner, subOp, nextJoiner);
+            if (nextJoiner.termType === 'Variable') {
+                joiner = nextJoiner;
+            }
```

Comment:
I don't think this statement is needed since it would only trigger in the last loop, after which `joiner` is not used anymore?

Reply:
I just did it to make the TS compiler happy, otherwise it complained that it can't assign an RDF object to RDF subject | variable.